src/Compilers/CSharp/Portable/Symbols/Metadata/PE/TupleTypeDecoder.cs

```diff
@@ -108,7 +110,8 @@ private static TypeSymbol DecodeTupleTypesInternal(TypeSymbol metadataType, Immu
                 var decoded = decoder.DecodeType(metadataType);
                 // If not all of the names have been used, the metadata is bad
                 if (!hasTupleElementNamesAttribute ||
-                    decoder._namesIndex == 0)
+                    decoder._namesIndex == 0 ||
+                    (object)decoded.VisitType(s_containsErrorPredicate, null) != null)
```

Comment:
>(object)decoded.VisitType(s_containsErrorPredicate, null) != null [](start = 20, length = 65)

I do not think this is the right way to fix the problem. We should be able to construct TupeTypeSymbol with the right names even when the definition of the underlying type is missing. It looks like the bug is in the ```private TypeSymbol DecodeType(TypeSymbol type)``` function below, it should call ```DecodeNamedType``` for ```SymbolKind.ErrorType``` and everything would just work as it should. 
In addition, we probably should improve error recovery in this function for other situations, i.e., if decoding failed and ```metadataType``` already has a use-site error associated with it, we should return ```metadataType``` instead of losing the original error information by returning UnsupportedMetadataTypeSymbol. I believe, this change alone, would make the new unit-test to pass, but the decoding should succeed in this scenario and the unit-test should be adjusted to verify that.

Reply:
@AlekseyTs I thought about making a `TupleTypeSymbol` with an underlying error type, but that felt riskier. I'll take another look and will push out to 15.7 depending on what I find.

As far as checking the use-site errors on `metadataType`, I think that is what I tried initially, but that created an unbounded recursion when compiling Roslyn with the bootstrap compiler. I didn't keep the precise stack trace.

Reply:
That worked out. Error types have everything to pass for ValueTuples. The main thing is to not mangle the name.